src/app/tests/suites/commands/system/SystemCommands.cpp

```diff
@@ -73,3 +78,38 @@ CHIP_ERROR SystemCommands::RunInternal(const char * command)
     VerifyOrReturnError(system(command) == 0, CHIP_ERROR_INTERNAL);
     return ContinueOnChipMainThread(CHIP_NO_ERROR);
 }
+
+CHIP_ERROR SystemCommands::CreateCommonCommandArgs(char * commandBuffer, size_t commandBufferSize, const char * scriptDir,
+                                                   const char * scriptName, uint16_t discriminator, uint16_t port, const char * kvs)
+{
+    memset(commandBuffer, 0, commandBufferSize);
+
+    // Reduce the buffer writer size by one to reserve the last byte for the null-terminator
+    chip::Encoding::BufferWriter writer(chip::Uint8::from_char(commandBuffer), commandBufferSize - 1);
```

Comment:
As I understand this, we use a binary writer and write strings and rely that the memset above did 0-initialize everything and then a 0 terminator is guaranteed.

This does sound a bit odd.

How about using StringBuilderBase ... it does exactly ensure null termination and builds strings and also has a fit method. It seems to use similar thing - a binary writer, but it is obviously for strings.

Reply:
https://github.com/project-chip/connectedhomeip/blob/master/src/lib/support/StringBuilder.h